### DISCUSSION TITLE
v18.2.0 android client build failed, but v18.0.0-dev01 build success

### DISCUSSION THREAD

**Discussion by thisleo** [alvr-org/ALVR#1001] [Q&A] [answered]
it is so strange，the version 18.2.0  run `cargo xtask prepare-deps --platform android` to build will show

Compiling alvr_filesystem v18.2.0 (/Users/Chendy/Downloads/Chrome/ALVR-master/alvr/filesystem)
   Compiling alvr_xtask v18.2.0 (/Users/Chendy/Downloads/Chrome/ALVR-master/alvr/xtask)
    Finished dev [unoptimized + debuginfo] target(s) in 3.12s
     Running `/Users/Chendy/Downloads/Chrome/ALVR-master/target/debug/alvr_xtask prepare-deps --platform android`

Wrong arguments.

cargo xtask
Developement actions for ALVR.

USAGE:
    cargo xtask <SUBCOMMAND> [FLAG] [ARGS]

SUBCOMMANDS:
    build-windows-deps  Download and compile external dependencies for Windows
    build-android-deps  Download and compile external dependencies for Android
    build-server        Build server driver, then copy binaries to build folder
    build-client        Build client, then copy binaries to build folder
    build-ffmpeg-linux  Build FFmpeg with VAAPI, NvEnc and Vulkan support. Only for CI
    publish-server      Build server in release mode, make portable version and installer
    publish-client      Build client for all headsets
    clean               Removes build folder
    kill-oculus         Kill all Oculus processes
    bump-versions       Bump server and client package versions
    clippy              Show warnings for selected clippy lints
    prettier            Format JS and CSS files with prettier; Requires Node.js and NPM.

FLAGS:
    --reproducible      Force cargo to build reproducibly. Used only for build subcommands
    --fetch             Update crates with "cargo update". Used only for build subcommands
    --release           Optimized build without debug info. Used only for build subcommands
    --experiments       Build unfinished features
    --nightly           Bump versions to nightly and build. Used only for publish subcommand
    --oculus-quest      Oculus Quest build. Used only for build-client subcommand
    --oculus-go         Oculus Go build. Used only for build-client subcommand
    --bundle-ffmpeg     Bundle ffmpeg libraries. Only used for build-server subcommand on Linux
    --no-nvidia         Additional flag to use with `build-server`. Disables nVidia support.
    --gpl               Enables usage of GPL libs like ffmpeg on Windows, allowing software encoding.
    --help              Print this text

ARGS:
    --version <VERSION> Specify version to set with the bump-versions subcommand
    --root <PATH>       Installation root. By default no root is set and paths are calculated using
                        relative paths, which requires conforming to FHS on Linux.

at v18.0.0-dev01 is all correct，how to solve this

**Comment by zmerp** [comment, marked as answer]
The master branch changed the build process. I made some changes in preparation to support future dependencies (x264) and future-proof in case we get a windows/linux client version.

**Reply by thisleo** [reply]
Can you tell me how to compile， thanks

**Reply by zmerp** [reply]
Just execute `cargo xtask build-client --release`.

**Reply by thisleo** [reply]
thanks，it‘s work！
